### PR TITLE
[fix] remove java.time.Duration to maintain minSdk support at version 21

### PR DESCRIPTION
**Pull request by suthar26**

### CODE REVIEW

File: android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/ConnectionHandler.java [hunk 1]
```diff
@@ -1,8 +1,7 @@
 package com.devcycle.sdk.android.eventsource;
 
-import java.time.Duration;
 
 interface ConnectionHandler {
-  void setReconnectionTime(Duration reconnectionTime);
+  void setReconnectionTime(long reconnectionTime);
```
Comment (nikolasleblanc) [inline android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/ConnectionHandler.java:5]: Does this have to be milliseconds now?
https://github.com/launchdarkly/okhttp-eventsource/compare/2.7.1...3.0.0#diff-9278fdbc0631a66e7cbc3b330a7e0d1d05f33a5590592c9f2754c72e5d762943R4

File: android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/EventParser.java [hunk 2]
```diff
@@ -227,7 +226,7 @@ public boolean processStream() throws IOException {
         break;
       case RETRY:
         if (DIGITS_ONLY.matcher(fieldValue).matches()) {
-          connectionHandler.setReconnectionTime(Duration.ofMillis(Long.parseLong(fieldValue)));
+          connectionHandler.setReconnectionTime((Long.parseLong(fieldValue)));
```
Comment (nikolasleblanc) [inline android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/EventParser.java:229]:
```suggestion
          connectionHandler.setReconnectionTime(Long.parseLong(fieldValue));
```

**Review by nikolasleblanc — COMMENTED** [review]
Looks like you've accounted for the milliseconds but just want to confirm, also that the TimeUnit in `(long, TimeUnit)` doesn't have to be explicitly provided as `null` the way it appears to be in a lot of the LD diff.

File: android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/ConnectionHandler.java (hunk 1 above)
Comment (suthar26) [reply android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/ConnectionHandler.java:5]: its out own connectionHandler and can be called whatever we want, I will add a comment that says it is Milliseconds

File: android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/EventSource.java [hunk 3]
```diff
@@ -67,29 +66,29 @@
...
+   * The default value for {@link Builder#maxReconnectTime(long)}: 30 seconds.
    */
-  public static final Duration DEFAULT_MAX_RECONNECT_TIME = Duration.ofSeconds(30);
+  public static final long DEFAULT_MAX_RECONNECT_TIME = 30 * 1000;
   /**
-   * The default value for {@link Builder#connectTimeout(Duration)}: 10 seconds.
+   * The default value for {@link Builder#connectTimeout(long)}: 10 seconds.
    */
-  public static final Duration DEFAULT_CONNECT_TIMEOUT = Duration.ofSeconds(10);
+  public static final long DEFAULT_CONNECT_TIMEOUT = 10*1000;
```
Comment (L-Joli) [inline android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/EventSource.java:79]:
```suggestion
  public static final long DEFAULT_CONNECT_TIMEOUT = 10 * 1000;
```

File: android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/EventSource.java [hunk 4]
```diff
@@ -67,29 +66,29 @@
...
   /**
-   * The default value for {@link Builder#writeTimeout(Duration)}: 5 seconds.
+   * The default value for {@link Builder#writeTimeout(long)}: 5 seconds.
    */
-  public static final Duration DEFAULT_WRITE_TIMEOUT = Duration.ofSeconds(5);
+  public static final long DEFAULT_WRITE_TIMEOUT = 5*1000;
```
Comment (L-Joli) [inline android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/EventSource.java:83]:
```suggestion
  public static final long DEFAULT_WRITE_TIMEOUT = 5 * 1000;
```

File: android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/EventSource.java [hunk 5]
```diff
@@ -67,29 +66,29 @@
...
   /**
-   * The default value for {@link Builder#readTimeout(Duration)}: 5 minutes.
+   * The default value for {@link Builder#readTimeout(long)}: 5 minutes.
    */
-  public static final Duration DEFAULT_READ_TIMEOUT = Duration.ofMinutes(5);
+  public static final long DEFAULT_READ_TIMEOUT = 5*60*1000;
```
Comment (L-Joli) [inline android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/EventSource.java:87]:
```suggestion
  public static final long DEFAULT_READ_TIMEOUT = 5 * 60 * 1000;
```

**Review by kaushalkapasi — APPROVED** [review]

File: android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/EventSource.java [hunk 6]
```diff
@@ -67,29 +66,29 @@
...
   /**
-   * The default value for {@link Builder#backoffResetThreshold(Duration)}: 60 seconds.
+   * The default value for {@link Builder#backoffResetThreshold(long)}: 60 seconds.
    */
-  public static final Duration DEFAULT_BACKOFF_RESET_THRESHOLD = Duration.ofSeconds(60);
+  public static final long DEFAULT_BACKOFF_RESET_THRESHOLD = 60*1000;
```
Comment (L-Joli) [inline android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/EventSource.java:91]:
```suggestion
  public static final long DEFAULT_BACKOFF_RESET_THRESHOLD = 60 * 1000;
```

File: android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/EventSource.java (hunk 3 above)
Comment (L-Joli) [reply android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/EventSource.java:79]: just to align with the format of `DEFAULT_MAX_RECONNECT_TIME`

**Review by nikolasleblanc — APPROVED** [review]

**Pull request merged**